imap/lmtp_sieve.c

```diff
@@ -1707,6 +1710,7 @@ static int sieve_keep(void *ac,
     script_data_t *sd = (script_data_t *) sc;
     deliver_data_t *mydata = (deliver_data_t *) mc;
     int ret = IMAP_MAILBOX_NONEXISTENT;
+    static unsigned mode = ACTION_KEEP;
```

Comment:
I'm a little surprised by this being `static`... is it because a single script run may execute 'keep' multiple times, and the behaviour of each depends on the behaviour of the previous, or something like that?

Reply:
Yep